angular/src/lib/toggle-switch/toggle-switch.component.ts

```diff
@@ -0,0 +1,102 @@
+import { Component, OnInit, Input, Output, EventEmitter, ChangeDetectorRef, forwardRef, HostBinding } from '@angular/core';
+import { NG_VALUE_ACCESSOR, ControlValueAccessor } from '@angular/forms';
+
+const CUSTOM_TOGGLE_SWITCH_VALUE_ACCESSOR: any = {
+  provide: NG_VALUE_ACCESSOR,
+  useExisting: forwardRef(() => ToggleSwitchComponent),
+  multi: true
+};
+
+@Component({
+  selector: 'cui-toggle-switch',
+  template: `
+    <input
+      class="cui-input cui-toggle-switch__input"
+      type="checkbox"
+      (change)="onSwitch($event)"
+      [attr.id]="htmlId"
+      [attr.name]="name"
+      [attr.tabindex]="tabIndex"
+      [checked]="checked"
+      [disabled]="disabled"
+    />
+    <label class="cui-toggle-switch__label" [attr.for]="htmlId">
+      <span class="cui-toggle-switch__label__container"></span>
+      <span class="cui-toggle-switch__label__text">{{label}}</span>
+    </label>
+  `,
+  styles: [],
+  providers: [CUSTOM_TOGGLE_SWITCH_VALUE_ACCESSOR]
+})
+export class ToggleSwitchComponent implements ControlValueAccessor {
+
+  /** @prop Optional CSS class name | '' */
+  @Input() class: string = '';
+  /** @prop Sets the attribute disabled to the ToggleSwitch | false */
+  @Input() disabled: boolean = false;
+  /** @prop Unique HTML ID used for tying label to HTML input for automated testing */
+  @Input() htmlId: string = '';
+  /** @prop ToggleSwitch label text | '' */
+  @Input() label: string = '';
+  /** @prop ToggleSwitch name for group | '' */
+  @Input() name: string = '';
+  /** @prop index of the ToggleSwitch in tab order */
+  @Input() tabIndex: number;
+  /** @prop String value that corresponds with ToggleSwitch button | '' */
+  @Input() value: any = '';
+
+  /** @prop Callback function invoked when user clicks the ToggleSwitch button | null */
+  @Output() onChange: EventEmitter<any> = new EventEmitter();
+
+  @HostBinding('class') get className(): string {
```

Comment:
Update this to the new pattern. I've added documentation in this PR #127

Reply:
updated to new pattern